rcl/test/rcl/test_events.cpp

```diff
@@ -72,22 +74,23 @@ class CLASSNAME (TestEventFixture, RMW_IMPLEMENTATION) : public ::testing::Test
     ret = rcl_node_init(this->node_ptr, name, "", this->context_ptr, &node_options);
     ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
     ts = ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, Strings);
+
+    default_qos_profile.history = RMW_QOS_POLICY_HISTORY_KEEP_LAST;
+    default_qos_profile.depth = 0;
+    default_qos_profile.reliability = RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT;
+    default_qos_profile.durability = RMW_QOS_POLICY_DURABILITY_SYSTEM_DEFAULT;
+    default_qos_profile.lifespan = {0, 0};
+    default_qos_profile.deadline = {DEADLINE_PERIOD_IN_S.count(), 0};
+    default_qos_profile.liveliness_lease_duration = {LIVELINESS_LEASE_DURATION_IN_S.count(), 0};
+    default_qos_profile.liveliness = RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC;
   }
 
-  rcl_ret_t setup_publisher(
-    const rmw_time_t & deadline,
-    const rmw_time_t & lifespan,
-    const rmw_time_t & liveliness_lease_duration,
-    const rmw_qos_liveliness_policy_t liveliness_policy)
+  rcl_ret_t setup_publisher(const rmw_qos_profile_t qos_profile)
```

Comment:
Modified this function to take a rmw_qos_profile_t instead of each qos policy separately. This helps with resuse for the new test that is being added.